### DISCUSSION TITLE
change addresses and channels on the same potentiometer?

### DISCUSSION THREAD

**Discussion by Nivolp1976** [tttapa/Control-Surface#577] [Q&A]
Is it possible to generate different selectors to change addresses and channels on the same potentiometer?

**Comment by Veroledez** [comment]
Hola, aqui hay algo similar que puedes adaptar para potenciometros

https://github.com/tttapa/Control-Surface/discussions/381

**Reply by Nivolp1976** [reply]
Thanks! CHANNEL_1 + 1 works, but I can't add from the loop, any ideas how to do it?

**Reply by Veroledez** [reply]
Hola, debes de copiar la clase de doble banco, copia y sube el codigo   que proporciona Peter, el que dice aqui el codigo completo y prueba 

https://github.com/tttapa/Control-Surface/discussions/381#discussioncomment-314391

**Reply by Nivolp1976** [reply]
I tried and it did not work with the master version yet to samd21. Thank you! I would like to know if there is a way to add +1 to the channel from the loop.

**Reply by Veroledez** [reply]
Usa la rama new input y nos dices que pasa

**Reply by tttapa** [reply]
> I tried and it did not work with the master version yet to samd21.

What exactly didn't work? I just tried it and it worked as expected. 

> Thank you! I would like to know if there is a way to add +1 to the channel from the loop.

If you're using a selector as in the example (https://github.com/tttapa/Control-Surface/discussions/381#discussioncomment-314391), use `selectors[0].increment()`.  
If you're using the bank directly, you can use `channelbank.select(channelbank.getSelection() + 1)`\*.

The `new-input` branch has been merged into master, you should now use the master branch instead.

<sub>(*) assuming of course that this would be a valid setting, you have to clamp or wrap around if necessary.</sub>

**Reply by Nivolp1976** [reply]
They are right, it works correctly on the master, what happened was the following: USBDebugMIDI_Interface midi; it was in the sample code.
I still can't change the channel in my simple code because it is a smart potentiometer.

`//SMART POTENTIOMETER

Bank<128> bank;

IncrementDecrementSelector<128> selector {bank, {9, 10}};

using  CCSmartPot = Bankable::CCSmartPotentiometer<128>;

CCSmartPot potentiometer {
  {bank, BankType::CHANGE_ADDRESS},     // Bank configuration
  A3,                                   
  {2, CHANNEL_1}, 
};`

**Reply by Veroledez** [reply]
Tendras    que crear una clase de potenciometro inteligente en la que mezcles el dualaddress
